sale_exception/models/sale.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2011 Akretion, Sodexis
+# Copyright 2018 Akretion
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, fields
+
+
+class ExceptionRule(models.Model):
+    _inherit = 'exception.rule'
+
+    model = fields.Selection(
+        selection_add=[
+            ('sale.order', 'Sale order'),
+            ('sale.order.line', 'Sale order line'),
+        ]
+    )
+    sale_ids = fields.Many2many(
+        'sale.order',
+        string="Sales")
+
+
+class SaleOrder(models.Model):
+    _inherit = ['sale.order', 'base.exception']
+    _name = 'sale.order'
+    _order = 'main_exception_id asc, date_order desc, name desc'
+
+    @api.model
+    def _exception_rule_eval_context(self, rec):
+        res = super(SaleOrder, self)._exception_rule_eval_context(rec)
```

Comment:
As it is python3, you can use `res = super()._exception_rule_eval_context(rec)` instead.